### PR TITLE
Merge v1.14.2

### PR DESCRIPTION
**Pull request by amsanghi**
fixes: NIT-2676

Merge upstream geth's v1.14.2 release into our geth fork:https://github.com/ethereum/go-ethereum/releases/tag/v1.14.2

nitro PR- https://github.com/OffchainLabs/nitro/pull/2552

# Testing done
geth and nitro tests pass

### CODE REVIEW

**Review by tsahee — CHANGES_REQUESTED** [review]
small comments + need to solve a few conflicts when merging with origin/master. When possible with conflicts - use code from 14.2

File: core/state/statedb.go [hunk 1]
```diff
@@ -38,8 +39,12 @@ import (
 	"github.com/ethereum/go-ethereum/trie/trienode"
 	"github.com/ethereum/go-ethereum/trie/triestate"
 	"github.com/holiman/uint256"
+	"golang.org/x/sync/errgroup"
 )
 
+// TriesInMemory represents the number of layers that are kept in RAM.
+const TriesInMemory = 128
```
[the remark this answers is not in the document]
Comment (amsanghi) [reply core/state/statedb.go]: Done.

File: core/state_transition.go [hunk 2]
```diff
@@ -523,8 +523,7 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	} else {
 		fee := new(uint256.Int).SetUint64(st.gasUsed())
 		fee.Mul(fee, effectiveTipU256)
-		st.state.AddBalance(tipReceipient, fee, tracing.BalanceIncreaseRewardTransactionFee)
-		tipAmount = fee.ToBig()
+		st.state.AddBalance(st.evm.Context.Coinbase, fee, tracing.BalanceIncreaseRewardTransactionFee)
```
[the remark this answers is not in the document]
Comment (amsanghi) [reply core/state_transition.go:526]: Done.

**Review by tsahee — CHANGES_REQUESTED** [review]
Minor comments.
Generally: I want to do a deeper review of how commit changed in 14.2 to think if we need more fixes but merge itself seems good.

File: core/state/statedb.go [hunk 3]
```diff
@@ -1284,33 +1342,15 @@ func (s *StateDB) Commit(block uint64, deleteEmptyObjects bool) (common.Hash, er
 		s.arbExtraData.activatedWasms = make(map[common.Hash]ActivatedWasm)
 	}
 
-	if codeWriter.ValueSize() > 0 {
-		if err := codeWriter.Write(); err != nil {
-			log.Crit("Failed to commit dirty codes", "error", err)
-		}
-	}
 	if wasmCodeWriter.ValueSize() > 0 {
```
[the remark this answers is not in the document]
Comment (amsanghi) [reply core/state/statedb.go]: Done.

**Review by tsahee — APPROVED** [review]
LGTM

**Pull request merged**